components/upProvider.tsx

```diff
@@ -44,30 +46,28 @@ export function useUpProvider() {
   return context;
 }
 
-  interface UpProviderProps {
+interface UpProviderProps {
   children: ReactNode;
 }
 
 export function UpProvider({ children }: UpProviderProps) {
-  const [provider] = useState(() =>
-    typeof window !== "undefined" ? createClientUPProvider() : null
-  );
-  const [client] = useState(() =>
-    typeof window !== "undefined" && provider
-      ? createWalletClient({
-          chain: luksoTestnet,
-          transport: custom(provider),
-        })
-      : null
-  );
-
   const [chainId, setChainId] = useState<number>(0);
   const [accounts, setAccounts] = useState<Array<`0x${string}`>>([]);
   const [contextAccounts, setContextAccounts] = useState<Array<`0x${string}`>>([]);
   const [walletConnected, setWalletConnected] = useState(false);
   const [selectedAddress, setSelectedAddress] = useState<`0x${string}` | null>(null);
   const [isSearching, setIsSearching] = useState(false);
 
+  const client: WalletClient | null = useMemo(() => {
```

Comment:
the more I look into the mini app templates using this inside of react lifecycle the less it makes sense to me. I think you can just create a function for this without the need to even memoize this. It can be a normal function. No useeffects or usememo. The less side effects the better. Just simple TS/JS function.
I found this tutorial about using Viem and Next but this can be applied to any react app. https://therg.dev/connect-a-wallet-with-viem-and-next/ Plus if we want to use hooks better to just use wagmi in our case its probably the easiest way